integration_test/install_test.go

```diff
@@ -105,6 +111,95 @@ func TestKrewInstall_Manifest(t *testing.T) {
 	test.AssertExecutableInPATH("kubectl-" + validPlugin)
 }
 
+func startServer() (*httptest.Server, error) {
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.RequestURI {
+		case "/konfig_localhost.yaml":
+			file, _ := os.Open("./testdata/konfig_localhost.yaml")
+			defer file.Close()
+			if _, err := io.Copy(w, file); err != nil {
+				http.Error(w, err.Error()+" - "+http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			}
+		case "/bundle.tar.gz":
+			file, _ := os.Open("./testdata/bundle.tar.gz")
+			defer file.Close()
+			if _, err := io.Copy(w, file); err != nil {
+				http.Error(w, err.Error()+" - "+http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			}
+		default:
+			if _, err := io.WriteString(w, http.StatusText(http.StatusNotFound)); err != nil {
+				http.Error(w, err.Error()+" - "+http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			}
+			w.WriteHeader(404)
+		}
+	}))
+
+	listener, err := net.Listen("tcp", LocalhostURL)
+	if err != nil {
+		return nil, errors.New("Trouble starting local server")
+	}
+	server.Listener = listener
+	server.Start()
+	return server, nil
+}
+
+func TestKrewInstall_ManifestURL(t *testing.T) {
+	skipShort(t)
+
+	server, err := startServer()
+	if err != nil {
+		t.Errorf("Trouble starting local server")
+	}
+	defer server.Close()
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	test.Krew("install",
+		"--manifest-url", LocalhostManifestURL).RunOrFail()
+	test.AssertExecutableInPATH("kubectl-" + validPlugin)
+}
+
+func TestKrewInstall_ManifestURLAndArchive(t *testing.T) {
+	skipShort(t)
+
+	server, err := startServer()
+	if err != nil {
+		t.Errorf("Trouble starting local server")
+	}
+	defer server.Close()
+
+	test, cleanup := NewTest(t)
+	defer cleanup()
+
+	err = test.Krew("install",
+		"--manifest-url", LocalhostManifestURL,
+		"--archive", filepath.Join("testdata", fooPlugin+".tar.gz")).Run()
+	if err == nil {
+		t.Errorf("Expected install to fail but was successful")
+	}
+}
+
+func TestKrewInstall_ManifestURLAndManifest(t *testing.T) {
+	skipShort(t)
+
+	server, err := startServer()
+	if err != nil {
+		t.Errorf("Trouble starting local server")
```

Comment:
you aren't logging what the error was. just t.Fatalf(error)

Also errorf doesn't stop the test. so the test will continue to execute.

Reply:
Done